request_test.go

```diff
@@ -0,0 +1,108 @@
+package profitbricks
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+const requestTestDataPath = "testdata"
+
+type SuiteRequest struct {
+	suite.Suite
+	client *Client
+	apiUrl string
+}
+
+func (s *SuiteRequest) SetupTest() {
+	s.client = NewClient("", "")
+	s.apiUrl = s.client.client.apiURL
+	httpmock.Activate()
+}
+
+func (s *SuiteRequest) TearDownTest() {
+	httpmock.DeactivateAndReset()
+}
+
+type SuiteWaitTillRequests struct {
+	SuiteRequest
+}
+
+func Test_WaitTillRequests(t *testing.T) {
+	suite.Run(t, new(SuiteWaitTillRequests))
+}
+
+func loadTestData(t *testing.T, filename string) []byte {
+	data, err := ioutil.ReadFile(filepath.Join(requestTestDataPath, filename))
+	assert.NoError(t, err)
+	return data
+}
+
+func (s *SuiteWaitTillRequests) Test_OK_NoSelector() {
+	listResponses := [][]byte{
+		loadTestData(s.T(), "request_request_till_no_request_matches_01.json"),
+		loadTestData(s.T(), "request_request_till_no_request_matches_02.json"),
```

Comment:
Why is the list of fixture requests so large? I would understand 1 queued + 1 non-queued request, but what are the other non-queued requests fore?

Reply:
rule of thumb

Reply:
what do you mean by that?

Reply:
I mean, that it looked legit to give a bit more than the actual required test data. But I don't want to argue about that, so here you go.